networksurvey/src/main/java/com/craxiom/networksurvey/NetworkSurveyActivity.java

```diff
@@ -71,6 +71,7 @@ public class NetworkSurveyActivity extends AppCompatActivity
         {
             // Since we are running on the Android 12+, we need to ask for the BLUETOOTH_CONNECT and BLUETOOTH_SCAN permissions
             PERMISSIONS = new String[]{
+                    Manifest.permission.CAMERA,
```

Comment:
Instead of requesting the camera permission when the main activity starts, it is best practice to request the permission when it is needed, especially since most users probably won't use the QR Code feature. The `BluetoothFragment#onResume()` method is a reasonable pattern you can follow, with the exception that you should request the permission when the QR Code button is pressed, and not in the onResume for the mqtt connection fragment. Let me know if you need help with this as I can work up some code for it.